linkup-cli/src/services/caddy.rs

```diff
@@ -1,167 +1,200 @@
+            .stdout(Stdio::null())
+            .stderr(Stdio::null())
+            .status()
+            .unwrap();
+
+        Command::new("caddy")
+            .args(["add-package", "github.com/pberkel/caddy-storage-redis"])
+            .stdout(Stdio::null())
+            .stderr(Stdio::null())
+            .status()
+            .unwrap();
+    }
 
-pub fn install_cloudflare_package() -> Result<()> {
-    Command::new("caddy")
-        .args(["add-package", "github.com/caddy-dns/cloudflare"])
-        .stdout(Stdio::null())
-        .stderr(Stdio::null())
-        .status()
-        .map_err(|err| CliError::StartCaddy(err.to_string()))?;
+    fn start(&self) -> Result<(), Error> {
+        log::debug!("Starting {}", Self::NAME);
 
-    Ok(())
-}
+        if std::env::var(LINKUP_CF_TLS_API_ENV_VAR).is_err() {
+            return Err(Error::MissingTlsApiTokenEnv);
+        }
 
-pub fn install_redis_package() -> Result<()> {
-    Command::new("caddy")
-        .args(["add-package", "github.com/pberkel/caddy-storage-redis"])
-        .stdout(Stdio::null())
-        .stderr(Stdio::null())
-        .status()
-        .map_err(|err| CliError::StartCaddy(err.to_string()))?;
+        let domains_and_subdomains: Vec<String> = self
+            .domains
+            .iter()
+            .map(|domain| format!("{domain}, *.{domain}"))
+            .collect();
 
-    Ok(())
-}
+        self.write_caddyfile(&domains_and_subdomains)?;
 
-fn write_caddyfile(domains: &[String]) -> Result<()> {
-    let mut redis_storage = String::new();
+        // Clear previous log file on startup
+        fs::write(&self.logfile_path, "")?;
 
-    if let Ok(redis_url) = std::env::var("LINKUP_CERT_STORAGE_REDIS_URL") {
-        // This is worth doing to avoid confusion while the redis storage module is new
-        check_redis_installed()?;
+        Command::new("caddy")
+            .current_dir(linkup_dir_path())
+            .arg("start")
+            .arg("--pidfile")
+            .arg(&self.pidfile_path)
+            .stdout(Stdio::null())
+            .stderr(Stdio::null())
+            .status()?;
+
+        Ok(())
+    }
+
+    pub fn stop(&self) -> Result<(), Error> {
+        log::debug!("Stopping {}", Self::NAME);
+
+        Command::new("caddy")
+            .current_dir(linkup_dir_path())
+            .arg("stop")
+            .stdout(Stdio::null())
+            .stderr(Stdio::null())
+            .status()?;
+
+        Ok(())
+    }
 
-        let url = url::Url::parse(&redis_url)
-            .map_err(|_| CliError::StartCaddy(format!("Invalid REDIS_URL: {}", redis_url)))?;
-        redis_storage = format!(
+    fn write_caddyfile(&self, domains: &[String]) -> Result<(), Error> {
+        let mut redis_storage = String::new();
+
+        if let Ok(redis_url) = std::env::var("LINKUP_CERT_STORAGE_REDIS_URL") {
+            // This is worth doing to avoid confusion while the redis storage module is new
```

Comment:
outdated comment?

Reply:
Removed in a8b3b09708ab573333cc77a78c433db621935323